### PR TITLE
Pushing 🅿️WM Improvements

### PR DESCRIPTION
**Pull request by AarC10**
- Optimized the PWM module by replacing float usages with uint32_t
- Microseconds are passed in to the setPeriod function now
- Duty Cycle percentage is now specified as an integer

### CODE REVIEW

File: include/EVT/io/PWM.hpp [hunk 1]
```diff
@@ -25,21 +25,21 @@ class PWM {
      *
      * @param[in] dutyCycle Duty cycle to set the pin to.
      */
-    virtual void setDutyCycle(float dutyCycle) = 0;
+    virtual void setDutyCycle(uint32_t dutyCycle) = 0;
 
     /**
      * Set the period for the PWM in seconds.
```
Comment (cbolles) [inline include/EVT/io/PWM.hpp:31]: Will probably need to change this comment to reflect the new units. True for other changed methods as well

File: src/io/platform/f3xx/PWMf3xx.cpp [hunk 2]
```diff
@@ -248,7 +248,7 @@ void PWMf3xx::setPeriod(float period) {
     // autoreload value into a valid range.
     do {
         prescaler++;
-        autoReload = period * clockFrequency / (prescaler + 1) - 1;
+        autoReload = clockFrequency / (prescaler + 1) - 1 / period;
```
Comment (cbolles) [inline src/io/platform/f3xx/PWMf3xx.cpp:251]: Now that we divide by the period here, we probably want to handle the case where someone enters 0 as a period

**Review by cbolles — CHANGES_REQUESTED** [review]
Looks good! Just a couple of changes

**Review by cbolles — APPROVED** [review]
Looks good!

**Review by mjmagee991 — APPROVED** [review]
Looks good

**Pull request merged**